README.md

```diff
@@ -29,6 +29,81 @@ gradle run -P args=”-p 1234 -d ./site”
 
 ###If you do not already have gradle installed on your machine:
 
-To run tests: gradlew test
+To run unit tests: gradlew test
+
+To run unit, and blackbox test: ./runAllTests.sh 
 
 To run the program: gradlew run
+
+----
+## Extending this Server into your own project 
+
+To create a jar file: gradle jar
+
+  The jar file should be created in ./build/libs/
+  
+This HttpServer can be extended by including the JAR file in your own project. The constructor for HttpServer takes two parameters, the port to run on and the directory the server should use. You will have the ability to create your own custom handlers and add them to the list of handlers the server can use. Your custom handler must implement HttpServer’s IHandler interface, which contains a single method `generateResponse()` which returns an instance of the class HttpResposne. 
+
+After you have created your own custom handler you can add a route to this handler by using the HttpServer’s method `addRoute(String path, String method, IHanlder handler)`
+
+Example of how to create an instance of the HttpServer, add a route to your custom handler and start the server.
+~~~~
```

Comment:
markdown tip:

you can use three backticks on a line on their own, put some code, then put three more backticks, and github will format them nicely for you.

You can even tell it that the following code is "java" or "bash" or whatever.

Here you go buddy
https://guides.github.com/features/mastering-markdown/#GitHub-flavored-markdown